mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -196,6 +218,76 @@ def _get_write_credential_infos(self, remote_file_paths):
             GetCredentialsForWrite, self.run_id, run_relative_remote_paths
         )
 
+    def download_trace(self, trace_id):
+        """
+        Downloads the trace data for the specified trace ID.
+
+        Args:
+            trace_id: The trace ID for which to download trace data.
+
+        Returns:
+            A dictionary containing the trace data.
+        """
+        cred = self._call_endpoint(
+            DatabricksMlflowArtifactsService,
+            GetCredentialsForTraceDataDownload,
+            None,
+            {"trace_id": trace_id},
+        )
+        signed_uri = cred.credential_info.signed_uri
+        headers = self._extract_headers_from_credentials(cred.credential_info.headers)
+        with tempfile.TemporaryDirectory() as temp_dir:
+            local_path = Path(temp_dir, "traces.json")
+            download_file_using_http_uri(
+                signed_uri,
+                local_path,
+                headers=headers,
+            )
+            with local_path.open("r") as f:
+                return json.load(f)
+
+    def upload_trace(self, trace: Trace):
```

Comment:
Also, is it a bit strange that `upload_trace` takes a `Trace` (should be `TraceData`) object, but `download_trace` returns a Python dictionary? IMO I think they should both take a dict. The client layer should be responsible for validating / transforming the TraceData, not the artifact repo.

Reply:
Updated the function signature